### PR TITLE
chore: move tabindex to the root slot of Spinner

### LINKED ISSUE #31025: [Bug]: Spinner focusing and screen readers issues
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 17.28 GB / 31.74 GB
  Browsers:
    Edge: Chromium (123.0.2420.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/edit/68jtch?file=src%2Fexample.tsx

### Bug Description

Run NVDA or JAWS 
Navigate to sample code:
1. https://stackblitz.com/edit/68jtch?file=src%2Fexample.tsx
2. set focus on test button 
3. Press TAB key navigate to the spinner 

## Actual Behavior
screen reader is silent

## Expected Behavior
screen reader should announce the progressbar element with corresponding aria-label

issue is that role and aria-label is on the different element like the tabindex itself: 
image
All attributes should be on the same element

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by tomi-msft**
Fixes #31025

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Avatar-mount_base.html">622</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Avatar-mount.html">640</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Button-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Field-mount_base.html">1099</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Field-mount.html">1133</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProvider-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProvider-mount.html">702</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">36</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">28</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/MakeStyles-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/MakeStyles-mount.html">884</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Persona-mount_base.html">1788</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/Persona-mount.html">1724</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/SpinButton-mount_base.html">1417</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31056/perf-test-react-components/SpinButton-mount.html">1404</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr> | `24.71 kB`<br />`8.35 kB` | `24.695 kB`<br />`8.346 kB` | `-15 B` <br />`-4 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `71.104 kB`<br />`20.52 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `219.611 kB`<br />`62.103 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.591 kB`<br />`14.356 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4f964dd8cc941caa781079ec29e39804f2d7cd7b'>4f964dd8cc941caa781079ec29e39804f2d7cd7b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-spinner/src/components/Spinner/useSpinner.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const useSpinner_unstable = (props: SpinnerProps, ref: React.Ref<HTMLElem
         // but since it would be a breaking change to fix it, we are casting ref to it's proper type
         ref: ref as React.Ref<HTMLDivElement>,
         role,
+        tabIndex,
```
Comment (smhigley) [inline packages/react-components/react-spinner/src/components/Spinner/useSpinner.tsx:32]: Might make sense to not extract (is that the word? 😅) `tabIndex` at all, and just let it be part of `...rest` now that it's going on the same place

**Review by smhigley — APPROVED** [review]
Just the one comment, otherwise looks good!

**Review by khmakoto — APPROVED** [review]

**Pull request merged**